File: backend/app/schemas/proposal.py
from typing import Optional
from uuid import UUID
from pydantic import BaseModel

class ProposalGenerateRequest(BaseModel):
    """Configuration for proposal generation."""
    include_cover: bool = True
    include_site_map: bool = True
    include_specs: bool = True
    include_energy: bool = True
    include_financials: bool = True
    include_equipment: bool = True

class ProposalTaskResponse(BaseModel):
    """Response when starting a generation task."""
    task_id: str
    status: str

class ProposalStatusResponse(BaseModel):
    """Status of the generation task."""
    task_id: str
    status: str # PENDING, STARTED, SUCCESS, FAILURE
    result_url: Optional[str] = None
    error: Optional[str] = None

File: frontend/src/test/fixtures/proposal.ts
export interface ProposalTaskResponse {
    task_id: string;
    status: 'PENDING' | 'STARTED' | 'SUCCESS' | 'FAILURE';
    result_url?: string;
    error?: string;
}

export type ProposalStatusResponse = ProposalTaskResponse;

export const mockProposalTaskResponse: ProposalTaskResponse = {
    task_id: 'task-123',
    status: 'PENDING'
};

export const mockProposalStatusPending: ProposalStatusResponse = {
    task_id: 'task-123',
    status: 'PENDING'
};

export const mockProposalStatusStarted: ProposalStatusResponse = {
    task_id: 'task-123',
    status: 'STARTED'
};

export const mockProposalStatusSuccess: ProposalStatusResponse = {
    task_id: 'task-123',
    status: 'SUCCESS',
    result_url: 'http://localhost/proposals/task-123.pdf'
};

export const mockProposalStatusFailure: ProposalStatusResponse = {
    task_id: 'task-123',
    status: 'FAILURE',
    error: 'PDF generation failed'
};
